src/Microsoft.DotNet.Interactive.Http.Tests/ParserTests.NamedRequest.cs

```diff
@@ -0,0 +1,233 @@
+﻿using System;
+using System.Linq;
+using FluentAssertions;
+using Microsoft.DotNet.Interactive.Http.Parsing;
+using Microsoft.DotNet.Interactive.Http.Parsing.Parsing;
+using Microsoft.DotNet.Interactive.Http.Tests.Utility;
+using Microsoft.DotNet.Interactive.Tests.Utility;
+using Xunit;
+using static Microsoft.DotNet.Interactive.Http.Tests.HttpParserTests;
+
+namespace Microsoft.DotNet.Interactive.Http.Tests;
+
+public partial class HttpParserTests
+{
+
+    public class NamedRequest
+    {
+        [Fact]
+        public void named_request_is_parsed_correctly()
+        {
+            var code = """
+                # @name login 
+
+                POST {{baseUrl}}/api/login HTTP/1.1 
+
+                Content-Type: application/x-www-form-urlencoded 
+
+
+
+                name=foo&password=bar 
+
+
+
+                ###
+                """;
+
+            var result = Parse(code);
+
+            result.SyntaxTree.RootNode.DescendantNodesAndTokens().OfType<HttpCommentNamedRequestNode>().Should().HaveCount(1);
+        }
+
+        [Fact]
+        public void named_request_can_reference_content_type_header()
+        {
+            var code = """
+                # @name login 
+
+                POST {{baseUrl}}/api/login HTTP/1.1 
+
+                Content-Type: application/x-www-form-urlencoded 
+
+
+
+                name=foo&password=bar 
+
+
+
+                ###
+                """;
+
+            var result = Parse(code);
+
+            var namedRequest = result.SyntaxTree.RootNode.DescendantNodesAndTokens().OfType<HttpCommentNamedRequestNode>().Single();
+
+            namedRequest.ValueNode.Text.Should().Be("login");
```

Comment:
The node type names could be clearer I think. I would think the "name node" should be referring to `login`. In other variable syntax nodes, the name node refers to the name of the variable, and the value refers to the value of the variable. 

A quick discussion to sort out naming might be useful.